php/class-plugin-base.php

```diff
@@ -141,20 +141,14 @@ public function locate_plugin() {
 		if ( '/' !== \DIRECTORY_SEPARATOR ) {
 			$file_name = str_replace( \DIRECTORY_SEPARATOR, '/', $file_name ); // Windows compat.
 		}
-		$plugin_dir = preg_replace( '#(.*plugins[^/]*/[^/]+)(/.*)?#', '$1', $file_name, 1, $count );
-		if ( 0 === $count ) {
```

Comment:
Yes. I don't see the point of forcing the plugin to be inside a `plugins` or `mu-plugins` directory. Although I can't imagine a situation where having a plugin elsewhere would actually be the case, enforcing this is simply not required.

I've removed the `preg_replace` in c48cd3af9849159155155dae235c17e8474db557 in favour of double `dirname`. This makes the assumption that the `plugin-base.php` file will remain in a child directory of the main plugin. I think that it's okay to make this assumption for now, especially if we will be pursuing the ideas proposed in #12 later on.